lib/improver/tests/ensemble_calibration/ensemble_calibration/helper_functions.py

```diff
@@ -38,9 +38,99 @@
 import iris
 from iris.coords import AuxCoord, DimCoord
 from iris.cube import Cube, CubeList
+from iris.tests import IrisTest
 import numpy as np
 
+from improver.tests.set_up_test_cubes import set_up_variable_cube
 from improver.utilities.cube_manipulation import concatenate_cubes
+from improver.utilities.warnings_handler import ManageWarnings
+
+
+IGNORED_MESSAGES = ["Collapsing a non-contiguous coordinate"]
+WARNING_TYPES = [UserWarning]
+
+
+class EnsembleCalibrationAssertions(IrisTest):
+
+    """Additional assertions, specifically for usage in the
+    ensemble calibration unit tests."""
+
+    def assertArrayAlmostEqualLowerPrecision(self, first, second, decimal=4):
+        """Overriding of the assertArrayAlmostEqual method to check whether
+        array are matching to 4 decimal places. This is justified
+        based on the default tolerance of the minimisation using the
+        Nelder-Mead algorithm of 0.0001, so that minimisations on different
+        machines would only be aiming to match to 4 decimal places.
+
+        Args:
+            first (np.array):
+                First array to compare.
+            second (np.array):
+                Second array to compare.
+            decimal (int):
+                Number of decimal places that will be included within the
+                comparison.
+         """
+        self.assertArrayAlmostEqual(first, second, decimal=decimal)
+
+    def assertAlmostEqualLowerPrecision(self, first, second, places=3):
+        """Overriding of the assertAlmostEqual method to check whether
+        array are matching to 3 decimal places. This is justified
+        based on a 0.0001 precision level difference within the
+        ensemble calibration coefficients that may escalate when these
+        coefficients are combined.
+
+        Args:
+            first (np.array):
+                First array to compare.
+            second (np.array):
+                Second array to compare.
+            places (int):
+                Number of decimal places that will be included within the
+                comparison.
+         """
+        self.assertAlmostEqual(first, second, places=places)
```

Comment:
I've amended these tests. so that the precision level that is fixed, so that it is not user configurable. Please see the associated notebook PR for a review of the precision level queries.